Created AutoScoreAndBalance

avakimc opened the pull request:
- sequence of commands that first scores the game piece, drives, and balances on the charge station
- created a void method in lift called scoreGamePiece() that is currently empty

avakimc committed:
Created AutoScoreAndBalance

- sequence of commands that first scores the game piece, drives, and balances on the charge station
- created a void method in lift called scoreGamePiece() that is currently empty

tails618 commented on src/main/java/frc/robot/commands/AutoScoreAndBalance.java:
```diff
@@ -0,0 +1,26 @@
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+public class AutoScoreAndBalance extends SequentialCommandGroup {
+    private PathPlannerTrajectory traj = PathPlanner.loadPath("Trajectory",
```
Make "Trajectory" a placeholder string variable, since we don't know what specific trajectory it is.

balghane commented on src/main/java/frc/robot/commands/AutoScoreAndBalance.java:
```diff
@@ -0,0 +1,26 @@
+import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+public class AutoScoreAndBalance extends SequentialCommandGroup {
```
Add comment for what this auto does

balghane replied:
Let's add a chooser to RobotContainer so we can choose which auto to run. See here: https://docs.wpilib.org/en/stable/docs/software/dashboards/smartdashboard/choosing-an-autonomous-program-from-smartdashboard.html

balghane commented on src/main/java/frc/robot/commands/AutoScoreAndBalance.java:
```diff
@@ -0,0 +1,26 @@
+            new PathConstraints(Calibrations.PLACEHOLDER_DOUBLE, Calibrations.PLACEHOLDER_DOUBLE));
+    public AutoScoreAndBalance(boolean isFirstPath, Lift lift, DriveSubsystem drive) {
+        addCommands(
+                new InstantCommand(lift::scoreGamePiece, lift),
```
Hm, it occurs to me this command will instantly finish, meaning the robot will start driving right away. We might want to do a RunCommand instead, and supply a function to check when the lift is done scoring.

`new RunCommand(lift::scoreGamePiece, lift).until(lift::doneScoring)`

That `doneScoring` would need to return a boolean

tails618 commented on src/main/java/frc/robot/commands/AutoScoreAndBalance.java:
```diff
@@ -0,0 +1,26 @@
+    public AutoScoreAndBalance(boolean isFirstPath, Lift lift, DriveSubsystem drive) {
+        addCommands(
+                new InstantCommand(lift::scoreGamePiece, lift),
+                drive.followTrajectoryCommand(traj, isFirstPath),
+                new AutoChargeStationBalance(drive));
```
We already have the charge station sequence, which drives up onto the charge station and balances. Either we should use that here (so have a path from scoring to *next to* the charge station), or scrap that sequence and just have every auton have a path to go from scoring/intaking to already on the charge station. We're never going to use the sequence in isolation because we won't have an auton that *only* balances but doesn't score or intake. @balghane thoughts?

tails618 replied:
(This is for lines 19 and 20.)

avakimc committed:
Revisions to AutoScoreAndBalance

- Added comments explaining what the autonomous does
- Added a chooser to RobotContainer that chooses which of the autonomous options to do

tails618 reviewed (approved):
Approved once you make those strings shorter

avakimc committed:
Made the descriptions of the autonomous options shorter

avakimc commented:
/format

github-actions[bot] committed:
Formatting fixes

avakimc committed:
Merge branch 'main' into ava/score-balance-auto

avakimc commented:
/format
